src/test/regress/expected/cte_inline_0.out

```diff
@@ -761,10 +761,10 @@ WITH fist_table_cte AS
     FROM
       fist_table_cte;
 DEBUG:  CTE fist_table_cte is going to be inlined via distributed planning
-DEBUG:  distributed statement: INSERT INTO cte_inline.test_table_1960000 AS citus_table_alias (key, value) SELECT key, value FROM (SELECT test_table.key, test_table.value, test_table.other_value FROM cte_inline.test_table_1960000 test_table) fist_table_cte WHERE ((worker_hash(key) OPERATOR(pg_catalog.>=) '-2147483648'::integer) AND (worker_hash(key) OPERATOR(pg_catalog.<=) '-1073741825'::integer))
-DEBUG:  distributed statement: INSERT INTO cte_inline.test_table_1960001 AS citus_table_alias (key, value) SELECT key, value FROM (SELECT test_table.key, test_table.value, test_table.other_value FROM cte_inline.test_table_1960001 test_table) fist_table_cte WHERE ((worker_hash(key) OPERATOR(pg_catalog.>=) '-1073741824'::integer) AND (worker_hash(key) OPERATOR(pg_catalog.<=) '-1'::integer))
-DEBUG:  distributed statement: INSERT INTO cte_inline.test_table_1960002 AS citus_table_alias (key, value) SELECT key, value FROM (SELECT test_table.key, test_table.value, test_table.other_value FROM cte_inline.test_table_1960002 test_table) fist_table_cte WHERE ((worker_hash(key) OPERATOR(pg_catalog.>=) 0) AND (worker_hash(key) OPERATOR(pg_catalog.<=) 1073741823))
-DEBUG:  distributed statement: INSERT INTO cte_inline.test_table_1960003 AS citus_table_alias (key, value) SELECT key, value FROM (SELECT test_table.key, test_table.value, test_table.other_value FROM cte_inline.test_table_1960003 test_table) fist_table_cte WHERE ((worker_hash(key) OPERATOR(pg_catalog.>=) 1073741824) AND (worker_hash(key) OPERATOR(pg_catalog.<=) 2147483647))
+DEBUG:  distributed statement: INSERT INTO cte_inline.test_table_1960000 AS citus_table_alias (key, value) SELECT key, value FROM (SELECT test_table.key, test_table.value, test_table.other_value FROM cte_inline.test_table_1960000 test_table) fist_table_cte WHERE (key IS NOT NULL)
+DEBUG:  distributed statement: INSERT INTO cte_inline.test_table_1960001 AS citus_table_alias (key, value) SELECT key, value FROM (SELECT test_table.key, test_table.value, test_table.other_value FROM cte_inline.test_table_1960001 test_table) fist_table_cte WHERE (key IS NOT NULL)
+DEBUG:  distributed statement: INSERT INTO cte_inline.test_table_1960002 AS citus_table_alias (key, value) SELECT key, value FROM (SELECT test_table.key, test_table.value, test_table.other_value FROM cte_inline.test_table_1960002 test_table) fist_table_cte WHERE (key IS NOT NULL)
+DEBUG:  distributed statement: INSERT INTO cte_inline.test_table_1960003 AS citus_table_alias (key, value) SELECT key, value FROM (SELECT test_table.key, test_table.value, test_table.other_value FROM cte_inline.test_table_1960003 test_table) fist_table_cte WHERE (key IS NOT NULL)
```

Comment:
this seems kind of ok, not worse than before, but it does make wonder whether we should be auto-creating not null constraints on the distribution column instead of adding this filter

Reply:
Even if we have a not null constraint on the distribution column values, a FULL JOIN can still yield NULL values for these columns. I think we should still keep some kind of filter if we wish to keep the current behaviour.

```sql
CREATE TABLE a(x int not null);
CREATE TABLE b(x int not null);
INSERT INTO a VALUES (1);
INSERT INTO b VALUES (2);
SELECT a.x, b.x FROM a FULL JOIN b USING (x);
┌────────┬────────┐
│   x    │   x    │
├────────┼────────┤
│      1 │ [NULL] │
│ [NULL] │      2 │
└────────┴────────┘
```

If we were to add not null constraints and remove these filters in question, Citus will behave similar to Vanilla PG. The following query fails on vanilla PG, but succeeds on Citus ( if a and b are distributed ) by silently ignoring null values.

```sql
INSERT INTO a SELECT a.x FROM a FULL JOIN b USING (x);
ERROR:  23502: null value in column "x" violates not-null constraint
DETAIL:  Failing row contains (null).
SCHEMA NAME:  public
TABLE NAME:  a
COLUMN NAME:  x
```